src/Compilers/Core/Portable/FileSystem/RelativePathResolver.cs

```diff
@@ -16,6 +16,8 @@ internal class RelativePathResolver : IEquatable<RelativePathResolver>
         public ImmutableArray<string> SearchPaths { get; }
         public string BaseDirectory { get; }
 
+        private ImmutableDictionary<(string reference, string baseFilePath), string> _cachedResults = ImmutableDictionary<(string reference, string baseFilePath), string>.Empty;
```

Comment:
@agocke What do you think here?  How long do `RelativePathResolvers` live?  Do we need some sort of eviction policy?